OpenTrons/opentrons-api
# fix(app): add loading state to protocol routes

closes RQA-1235 and RAUT-466


# Overview



This PR addresses loading state skeletons/fixes/refactoring to the protocol routes.

### ProtocolDetails Route
Adds new loading skeleton to ProtocolDetails route. Adds a spinner to the Start Setup when clicked.

**New Skeleton**

![protocolDetailsSkeleton](https://github.com/Opentrons/opentrons/assets/64858653/8d07d2d0-44cd-4252-9110-e79b11b9592d)
**Make Setup Spinner**
![makeSetupSpinner](https://github.com/Opentrons/opentrons/assets/64858653/f6bbd82e-af1a-405a-9150-28802c85acec)
### ProtocolSetup
Fixes jitter/styling errors with the ProtocolSetup route skeleton to better match existing page components and adds missing testing to the skeleton. Fixes ProtocolSetupStep button sizing issues when enabled/disabled.
**Updated Skeleton**
![updated skeleton](https://github.com/Opentrons/opentrons/assets/64858653/11c0cfb4-b558-45c4-8033-f1c73537e1f7)
**Consistent Button Sizing**
![sizing](https://github.com/Opentrons/opentrons/assets/64858653/560ea3b8-27c3-41d3-bc0e-2014493d0ce1)

# Test Plan



ProtocolDetails Skeleton
1. Upload a new protocol and click it (ProtocolDetails route).
2. If loading occurs, there should be a skeleton, and the Make Setup button should be disabled.
3. After loading completes, the skeleton should disappear and the make setup button should be selectable.
4. Click Make Setup - there should be a spinner until the redirect occurs.

ProtocolSetup Skeleton
1. If any loading occurs on this page, there should be a skeleton. No jittering should occur. 
2. Disabled setup buttons should be the same size as enabled buttons.

# Changelog



- Created ProtocolDetailsSkeleton and testing.
- ProtocolSetupSkeleton refactored to own file and testing added.
- Refactored props within ProtocolDetails to interfaces to promote consistency.
- Added loading spinner on Make Setup button.
- Added a transparent icon in ProtocolSetup for the ProtocolSetupStep buttons.

# Review requests



# Risk assessment
